domains/certificates/Query.API/API/DataSyncSyncer/Startup.cs

```diff
@@ -1,9 +1,18 @@
+using System;
+using API.DataSyncSyncer.Configurations;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 
 namespace API.DataSyncSyncer;
 
 public static class Startup
 {
-    public static void AddDataSyncSyncer(this IServiceCollection services)
-        => services.AddHostedService<DataSyncSyncerWorker>();
+    public static void AddDataSyncSyncer(this IServiceCollection services, IConfiguration configuration)
+    {
+        var datasyncUrl = configuration.GetSection(DatasyncOptions.Datasync).Value;
+        services.AddHttpClient<DataSyncService>(client => client.BaseAddress = new Uri(datasyncUrl));
+
+        services.AddTransient<DataSyncService>();
```

Comment:
Just a note: DataSyncService is lifetime "transient". If DataSyncService holds state, then that lifetime is a problem.